Makefile

```diff
@@ -68,15 +68,25 @@ COVERAGE_PACKAGES=$(shell go list ./... | \
 
 COVERAGE_PACKAGES_LIST=$(shell echo $(COVERAGE_PACKAGES) | tr " " ",")
 
+CLI_SOURCES=$(shell find cli client apis pkg -name "*.go" | grep -v "_test.go")
+
+DAEMON_SOURCES=$(shell find lxcfs credential apis test plugins network ctrd internal \
```

Comment:
maybe we can use `go list ./...` to get all the related packages.

but, one question for this change, do we have any user cases about this change?

Reply:
This is used to detect source files changed.

$ touch main.go

It'll build pouchd only.

$ make 
build-daemon: bin/pouchd

Reply:
Source files as target dependencies

bin/${CLI_BINARY_NAME}: $(CLI_SOURCES)

Reply:
well, I think we should use `go build` to handle this. In the go.10, the go build can detect the out -of-date package. 

```
# https://golang.org/doc/go1.10#build
The go build command now detects out-of-date packages purely based on the content of source files, specified build flags, and metadata stored in the compiled packages. Modification times are no longer consulted or relevant. The old advice to add -a to force a rebuild in cases where the modification times were misleading for one reason or another (for example, changes in build flags) is no longer necessary: builds now always detect when packages must be rebuilt. (If you observe otherwise, please file a bug.)
```

Reply:
I have tested with golang v1.10.4

$ go version
go version go1.10.4 linux/amd64

============old version===================
$ time make -f Makefile.old
build volume modules
build-daemon: bin/pouchd
build-cli: bin/pouch

real	0m16.116s
user	0m3.324s
sys	0m11.312s

$ touch cli/cli.go
$ time make -f Makefile.old
build volume modules
build-daemon: bin/pouchd
build-cli: bin/pouch

<b>real	0m17.254s</b>
user	0m3.452s
sys	0m11.112s

$ touch main.go
$ time make -f Makefile.old
build volume modules
build-daemon: bin/pouchd
build-cli: bin/pouch

real	0m16.281s
user	0m3.408s
sys	0m11.296s

===========new version=================
$ touch cli/cli.go

$ time make -f Makefile.new
build-cli:  bin/pouch

<b>real	0m2.333s</b>
user	0m0.348s
sys	0m1.648s

$ touch main.go
$ time make -f Makefile.new
build-daemon: bin/pouchd

real	0m15.283s
user	0m3.304s
sys	0m10.316s

Reply:
@manlge thank you for the testing. 

If we manage file change manually, we need to pay attention to the new package or new go source file. Maybe the `grep` can handle all the cases. But the go complier can handle this well. For this reason, I don't think we need to change it.

BTW, I also provide my test result.

```
# vagrant @ ubuntu-xenial in ~/go/src/github.com/alibaba/pouch on git:master o [9:34:48]
$ time make
build volume modules
build-daemon: bin/pouchd
build-cli: bin/pouch
make  13.63s user 6.40s system 178% cpu 11.245 total

# vagrant @ ubuntu-xenial in ~/go/src/github.com/alibaba/pouch on git:master o [9:37:55]
$ time make
build volume modules
build-daemon: bin/pouchd
build-cli: bin/pouch
make  4.04s user 1.38s system 118% cpu 4.573 total

# vagrant @ ubuntu-xenial in ~/go/src/github.com/alibaba/pouch on git:master o [9:38:04]
$ go version
go version go1.10.3 linux/amd64

# vagrant @ ubuntu-xenial in ~/go/src/github.com/alibaba/pouch on git:master o [9:38:18]
$ touch main.go

# vagrant @ ubuntu-xenial in ~/go/src/github.com/alibaba/pouch on git:master o [9:40:15]
$ time make
build volume modules
build-daemon: bin/pouchd
build-cli: bin/pouch
make  3.97s user 1.43s system 120% cpu 4.468 total
```